tests/functional_test.py

```diff
@@ -1,88 +1,167 @@
-#!/usr/bin/env python3
-
 """
-./speculos.py --log-level automation:DEBUG --automation file:$HOME/app-xrp/tests/automation.json ~/app-xrp/bin/app.elf &
+./speculos.py --log-level automation:DEBUG ~/app-xrp/bin/app.elf &
 
 export LEDGER_PROXY_ADDRESS=127.0.0.1 LEDGER_PROXY_PORT=9999
 pytest-3 -v -s
 """
-import os
+from pathlib import Path
+import json
+from hashlib import sha512, sha256
 import pytest
-import pathlib
-from xrp import XRPClient, DEFAULT_PATH
-from ledgerwallet.params import Bip32Path
+from Crypto.Hash import RIPEMD160
+from ledgerwallet.params import Bip32Path  # type: ignore [import]
 from ragger.backend import RaisePolicy
+from ragger.bip import calculate_public_key_and_chaincode, CurveChoice
 from ragger.error import ExceptionRAPDU
+from .xrp import XRPClient, Errors
+from .utils import DEFAULT_PATH, DEFAULT_BIP32_PATH, util_navigate
+
+
+APP_VERSION = "2.3.2"
```

Comment:
Nice to have : parse the version from the makefile so you don't need to update this file when updating the version.

Reply:
Done, within a `verify_version` function in `utils` module